tests/test_transit_api.py

```diff
@@ -339,6 +340,27 @@ def test_8_signing_with_subkeys(self):
       r = self.openpgp.verify_signed_data(fixed_name, base64_input, signature=s)
       self.assertFalse(r['data']['valid'])
 
+  def test_9_expiring_master_keys(self):
+    KEY_EXPIRES_SECS = 20
+    fixed_input = 'Hello, world!'
+    base64_input = self.base64ify(fixed_input)
+
+    for key_type in ALLOWED_KEY_TYPES:
+      fixed_name = self.random_name()
+
+      # Create an expiring signing subkey.
+      self.openpgp.create_key(fixed_name, key_type=key_type, expires=KEY_EXPIRES_SECS)
+
+      # Test key expiration.
+      r = self.openpgp.sign_data(fixed_name, base64_input)
+      s = r['data']['signature']
+      r = self.openpgp.verify_signed_data(fixed_name, base64_input, signature=s)
+      self.assertTrue(r['data']['valid'])
+      # Sleep until key expires.
+      time.sleep(KEY_EXPIRES_SECS)
```

Comment:
Every time you `time.sleep` for 20 seconds during an automated test, a baby angel sheds a tear ;-)

But hey, if it works, I don't mind.